diffkemp/semdiff/result.py

```diff
@@ -109,3 +108,60 @@ def report_stat(self, show_errors=False):
                     if r.kind == Result.Kind.ERROR:
                         print(f)
                 print()
+
+    def report_object_stat(self):
+        """
+        Report detailed statistics about compared objects.
+        Prints the total count of unique non-equal objects (inner diffs)
+        excluding those covered by syntax diffs.
+        Also prints the count and percentage of those which are function diffs,
+        macro diffs, inline asm diffs and empty diffs.
+        """
+        # Wrapper class to store in set
+        class UniqueDiff:
+            def __init__(self, res):
+                self.res = res
+
+            def __eq__(self, other):
+                return self.res.first.name == other.res.first.name
+
+            def __hash__(self):
+                return hash(self.res.first.name)
+
+        # Convert inner result to set of unique diffs
+        unique_diffs = set()
+        for _, inner_res_out in self.inner.items():
+            for _, inner_res in inner_res_out.inner.items():
+                if (inner_res.diff == "" and
+                        inner_res.first.covered_by_syn_diff):
+                    continue
+                unique_diffs.add(UniqueDiff(inner_res))
+
+        # Generate counts
+        total = len(unique_diffs)
+        functions = len([r for r in unique_diffs
+                         if not r.res.first.is_syn_diff])
+        macros = len([r for r in unique_diffs
+                      if (r.res.first.is_syn_diff and
+                          not r.res.first.name.startswith("assembly code"))])
+        asm = len([r for r in unique_diffs
+                  if (r.res.first.is_syn_diff and
+                      r.res.first.name.startswith("assembly code"))])
+        empty = len([r for r in unique_diffs if r.res.diff == ""])
+
+        # Print statistics
+        print("Total differences:  {}".format(total))
+        print("In functions:       {0} ({1:.0f}%)".format(functions,
+              functions / total * 100))
+        print("In macros:          {0} ({1:.0f}%)".format(macros,
+              macros / total * 100))
+        print("In inline assembly: {0} ({1:.0f}%)".format(asm,
```

Comment:
"In inline assemblies". Let's use plural everywhere.

Reply:
Actually neither "inline assembly" or "inline assemblies" is correct (the first should be "inline assembly code", because "assembly" refers to the **process**, not to the result; the second one is grammatically incorrect, because "assembly" in this sense is uncountable).

Reply:
Agreed, inline assembly code is correct.